include/CLI/Option.hpp

```diff
@@ -427,15 +427,15 @@ class Option : public OptionBase<Option> {
     }
 
     /// Adds a transforming Validator with a built in type name
-    Option *transform(Validator Validator, std::string Validator_name = "") {
+    Option *transform(Validator Validator, const std::string &Validator_name = "") {
         validators_.insert(validators_.begin(), std::move(Validator));
         if(!Validator_name.empty())
             validators_.front().name(Validator_name);
         return this;
     }
 
     /// Adds a Validator-like function that can change result
-    Option *transform(std::function<std::string(std::string)> func,
+    Option *transform(const std::function<std::string(std::string)> &func,
```

Comment:
I have gone back and forth on whether std::function should be taken by value and moved or taken by constref and copied.  I think I usually default to take by value since a very common usages is to give a lambda function in the argument, but I am not entirely sure what is best here.

Reply:
actually in this case constref is probably best since we have to do a copy in the internal lambda (at least until C++14 is the minimum mode in CLI11)

Reply:
Probably you're right. It won't make a huge difference I guess. And with this PR it is improved compared to before, so I suggest keeping it.